src/helpers/utils.js

```
@@ -0,0 +1,27 @@
+const API_KEY = "740fee2f337ae281a8894434d230cfd0";
+
+export const fetchWeather = (cityName, units) => {
+  return fetch(
+    `http://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=${units}&appid=${API_KEY}`,
+  )
+    .then(response => response.json())
+    .then(data => {
+      return data;
+    })
+    .catch((error) => {
+      console.log(error)
+    })
+};
+
+export const fetchForecast = (cityName, units) => {
+  return fetch(
+    `http://api.openweathermap.org/data/2.5/forecast?q=${cityName}&units=${units}&appid=${API_KEY}`,
+  )
+    .then(response => response.json())
+    .then(data => {
+      return data;
+    })
+    .catch((error) => {
+      console.log(error)
+    })
+};
```

How would you refactor this to handle errors on the UI? How would you send the error and what would you do with it?

---

I would return the error message from the util function and in WeatherContainer create a new state value for identifying there is an error, and passing the error message to the UI.

```
  if (response.ok) {
    return response;
  } else {
    throw new Error('Something went wrong. Please try your search again);
  }
})
    .then(response => response.json())
    .then(data => {
      return data;
    })
.catch((error) => {
      return error
    })
```

when rendering UI
```
{error && (
<AlertComponent alertMessage={error} />
```